src/java.desktop/share/classes/javax/swing/filechooser/FileSystemView.java

```diff
@@ -255,6 +256,62 @@ public Icon getSystemIcon(File f) {
         }
     }
 
+   /**
+    * Returns an icon for a file, directory, or folder as it would be displayed
+    * in a system file browser for the requested size.
+    * <p>
+    * Example: <pre>
+    *     FileSystemView fsv = FileSystemView.getFileSystemView();
+    *     Icon icon = fsv.getSystemIcon(new File("application.exe"), 64, 64);
+    *     JLabel label = new JLabel(icon);
+    * </pre>
+    *
+    * @implSpec The available icons may be platform specific and so the
+    * available sizes determined by the platform. Therefore an exact match
+    * for the requested size may not be possible.
+    *
+    * The icon returned may be a multi-resolution icon image,
+    * which allows better support for High DPI environments
+    * with different scaling factors.
+    *
+    * @param f a {@code File} object for which the icon will be retrieved
+    * @param width width of the icon in user coordinate system.
+    * @param height height of the icon in user coordinate system.
+    * @return an icon as it would be displayed by a native file chooser
+    * or null if invalid parameters are passed such as reference to a
+    * non-existing file.
```

Comment:
non-existent not non-existing, but I think null deserves an IAE - as I had written a couple of days ago.

I see you dropped the words about null if the size is too large.
Should I interpret that as meaning one of the things I suggested - that this is just another case of a closest match ?

Reply:
Thanks for correction. Yes - we will return the best match no matter how big the requested size is.

Fixed IAE in case of null file.